lib/collection/src/collection_manager/holders/segment_holder.rs

```diff
@@ -753,7 +753,7 @@ impl<'s> SegmentHolder {
         };
 
         // Create temporary appendable segment to direct all proxy writes into
-        let tmp_segment = LockedSegment::new(build_segment(collection_path, &config, false)?);
+        let tmp_segment = LockedSegment::new(build_segment(segments_path, &config, false)?);
```

Comment:
Most importantly, we built new segments in the collection path before, instead of the segments path.